### ISSUE TITLE
logProvider Issue: No logging from winston when a call is given

### ISSUE THREAD

**Issue by Spenstar** [chimurai/http-proxy-middleware#194]
### Behavior without logProvider override
```console
[HPM] Rewriting path from "/some_path/request" to "/api/request"
[HPM] GET /some_path/api/request ~> http://some_url.com:port
```

### Expected behavior
```console
TIME_LOGGED - info: [HPM] Rewriting path from "/some_path/request" to "/api/request"
TIME_LOGGED - info: [HPM] GET /some_path/api/request ~> http://some_url.com:port
```
### Actual behavior
No output to console

### Setup

* http-proxy-middleware: `0.17.4`
* server: express `4.14.0`
* logger: winston `2.3.1`

#### proxy middleware configuration / Server Mounting
```javascript
var http_proxy = require('http-proxy-middleware');
var express = require('express');
var router = express.Router();

var requestTime = function (req, res, next) {
  req._startTime = new Date().getTime();
  next();
};

router.use('/some_path', requestTime, proxy('http://some_url.com', '/api'));

function onProxyRequest(path) {
  return function (proxyReq, req) {
    var values = serviceCallerId(req);
    proxyReq.setHeader('User-Agent', values.userAgent);
    proxyReq.setHeader('X-Orig-Base', '/some/api' + path);
    if (!req.headers.accept) {
      proxyReq.setHeader('accept', 'application/json');
    }
    return proxyReq;
  };
}

function proxy(url, path) {
  var pathRewrite = {};
  pathRewrite['^' + path + '/'] = '/';
  var options = {
    logLevel: 'debug',
    logProvider: function (provider) {
      var winston = require('winston');
      return new (winston.Logger)({
        transports: [
          new (winston.transports.Console)(
              {'timestamp':true})
        ]
      });
    },
    target: url,
    changeOrigin: true,
    pathRewrite: pathRewrite,
    onProxyReq: onProxyRequest(path)
  };
  return http_proxy(options);
}
```

**Comment by alexduros** [comment]
I faced the same issue. 
You have to setup your winston logger instance with the same logLevel that you setup in HPM side.
Actually, for debug logLevel, HPM use `logger.debug` method and winston default logLevel is `info` so you have to add `logLevel: 'debug'` option to your winston instance.

**Comment by chimurai** [comment]
fixed in 3.0.0-beta

https://github.com/chimurai/http-proxy-middleware/blob/master/recipes/logger.md
